service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/CoinRedemptionService.kt

```diff
@@ -0,0 +1,65 @@
+package io.provenance.digitalcurrency.consortium.service
+
+import io.provenance.digitalcurrency.consortium.api.DepositFiatRequest
+import io.provenance.digitalcurrency.consortium.bankclient.BankClient
+import io.provenance.digitalcurrency.consortium.config.BankClientProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.CoinBurnRecord
+import io.provenance.digitalcurrency.consortium.domain.CoinRedemptionRecord
+import io.provenance.digitalcurrency.consortium.domain.TxStatus
+import io.provenance.digitalcurrency.consortium.extension.coinsAmount
+import io.provenance.digitalcurrency.consortium.extension.findWithdrawEvent
+import io.provenance.digitalcurrency.consortium.extension.isSingleTx
+import org.springframework.stereotype.Service
+
+@Service
+class CoinRedemptionService(
+    private val bankClient: BankClient,
+    private val pbcService: PbcService,
+    bankClientProperties: BankClientProperties,
+    private val pbcTimeoutService: PbcTimeoutService
+) {
+    private val log = logger()
+    private val bankDenom = bankClientProperties.denom
+
+    fun createEvent(coinRedemptionRecord: CoinRedemptionRecord) {
```

Comment:
was the idea keeping redeem unbatched for now? 

wondering if we can just stop the CoinBurnRecord from being auto inserted but require bank middleware to call in to burn going forwards

Reply:
yes - unless I'm overthinking, it'd be a mess to handle multiple redemptions for the same amount in one batch. If we decouple the burns completely from this process I think that would make it more feasible to batch them. I can come back to this after I get the logic in the usdf since I have to fix it up anyway.

Reply:
If we update the SC to include more detailed attributes for how much bank coin redeemed, it would be a lot simpler to parse in a batch but yeah there’s some changes needed here